src/SignalR/clients/cpp/src/signalrclient/hub_connection_impl.cpp

```diff
@@ -54,14 +54,27 @@ namespace signalr
         // weak_ptr prevents a circular dependency leading to memory leak and other problems
         auto weak_hub_connection = std::weak_ptr<hub_connection_impl>(this_hub_connection);
 
-        m_connection->set_message_received_json([weak_hub_connection](const web::json::value& message)
+        m_connection->set_message_received([weak_hub_connection](const utility::string_t& message)
         {
             auto connection = weak_hub_connection.lock();
             if (connection)
             {
                 connection->process_message(message);
             }
         });
+
+        m_connection->set_disconnected([weak_hub_connection]()
+        {
+            auto connection = weak_hub_connection.lock();
+            if (connection)
+            {
+                connection->m_handshakeTask.set_exception(signalr_exception(_XPLATSTR("connection closed while handshake was in progress.")));
+                if (connection->m_disconnected)
```

Comment:
You don't need to null check m_disconnected. Just add `m_disconnected([](){})` to the hub_connection_impl initializer list like it is in connection_impl.cpp.